Library/Homebrew/tap.rb

```diff
@@ -15,12 +23,18 @@ def initialize(user, repo, remote=nil)
     @repo = repo
     @name = "#{@user}/#{@repo}".downcase
     @path = TAP_DIRECTORY/"#{@user}/homebrew-#{@repo}".downcase
-    if installed?
+    @custom_remote = remote
+  end
+
+  # The remote path to this {Tap}.
+  # e.g. `https://github.com/user/homebrew-repo`
+  def remote
+    @remote ||= if (@path/".git").exist?
       @path.cd do
-        @remote = Utils.popen_read("git", "config", "--get", "remote.origin.url").chomp
+        Utils.popen_read("git", "config", "--get", "remote.origin.url").chomp
       end
     else
-      @remote = remote || "https://github.com/#{@user}/homebrew-#{@repo}"
+      @custom_remote || "https://github.com/#{@user}/homebrew-#{@repo}"
```

Comment:
I guess it's used by `homebrew-bundle` for dumping etc.? Hmm. Could maybe return `nil` or raise an exception but either approach will need special-cased in `homebrew-bundle`.